dpp/src/data_contract/state_transition/data_contract_update_transition/validation/state/validate_data_contract_update_transition_state_factory.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{
+    data_contract::DataContract, errors::consensus::basic::BasicError, mocks,
+    state_repository::StateRepositoryLike, validation::ValidationResult,
+};
+use anyhow::Result;
+
+pub struct ValidateDataContractUpdateTransitionStateFactory<SR>
```

Comment:
I would rename this to something else, as you're creating an instance of factory, which is not actually the case - you're creating the validator. I'd suggest naming it as something like `DataContractUpdateTransitionStateValidator`

Reply:
good point, fixed